src/Categories/Category/Monoidal/Interchange.agda

```diff
@@ -0,0 +1,603 @@
+
+-- An abstract definition of an interchange map with the minimal set
+-- of coherence laws required to make the tensor product ⊗ of C a
+-- monoidal functor.  (See also Categories.Functor.Monoidal.Tensor.)
+
+record HasInterchange (M : Monoidal C) : Set (o ⊔ ℓ ⊔ e) where
+  open Monoidal M
+  open MonoidalUtilities.Shorthands M
+
+  -- The "four middle interchange" for tensor products.
+
+  field swapInner : ∀ {W X Y Z} → (W ⊗₀ X) ⊗₀ (Y ⊗₀ Z) ≅ (W ⊗₀ Y) ⊗₀ (X ⊗₀ Z)
+
+  module swapInner {W X Y Z} = _≅_ (swapInner {W} {X} {Y} {Z})
+  private
+    i⇒ = swapInner.from
+    i⇐ = swapInner.to
+
+  -- Naturality and coherence laws of the interchange.
+
+  field
+
+    swapInner-natural : ∀ {W₁ X₁ Y₁ Z₁ W₂ X₂ Y₂ Z₂}
+                        {f : W₁ ⇒ W₂} {g : X₁ ⇒ X₂}
+                        {h : Y₁ ⇒ Y₂} {i : Z₁ ⇒ Z₂} →
+                        i⇒ ∘ (f ⊗₁ g) ⊗₁ (h ⊗₁ i) ≈ (f ⊗₁ h) ⊗₁ (g ⊗₁ i) ∘ i⇒
+
+    swapInner-assoc : ∀ {X₁ Y₁ Z₁ X₂ Y₂ Z₂} →
+                      [ ((X₁ ⊗₀ X₂) ⊗₀ (Y₁ ⊗₀ Y₂)) ⊗₀ (Z₁ ⊗₀ Z₂) ⇒
+                        (X₁ ⊗₀ (Y₁ ⊗₀ Z₁)) ⊗₀ (X₂ ⊗₀ (Y₂ ⊗₀ Z₂)) ]⟨
+                        i⇒ ⊗₁ id   ⇒⟨ ((X₁ ⊗₀ Y₁) ⊗₀ (X₂ ⊗₀ Y₂)) ⊗₀ (Z₁ ⊗₀ Z₂) ⟩
+                        i⇒         ⇒⟨ ((X₁ ⊗₀ Y₁) ⊗₀ Z₁) ⊗₀ ((X₂ ⊗₀ Y₂) ⊗₀ Z₂) ⟩
+                        α⇒ ⊗₁ α⇒
+                      ≈ α⇒         ⇒⟨ (X₁ ⊗₀ X₂) ⊗₀ ((Y₁ ⊗₀ Y₂) ⊗₀ (Z₁ ⊗₀ Z₂)) ⟩
+                        id ⊗₁ i⇒   ⇒⟨ (X₁ ⊗₀ X₂) ⊗₀ ((Y₁ ⊗₀ Z₁) ⊗₀ (Y₂ ⊗₀ Z₂)) ⟩
+                        i⇒
+                      ⟩
+
+    swapInner-unitˡ : ∀ {X Y} →
+                      [ unit ⊗₀ (X ⊗₀ Y) ⇒ (X ⊗₀ Y) ]⟨
+                        λ⇐ ⊗₁ id   ⇒⟨ (unit ⊗₀ unit) ⊗₀ (X ⊗₀ Y) ⟩
+                        i⇒         ⇒⟨ (unit ⊗₀ X) ⊗₀ (unit ⊗₀ Y) ⟩
+                        λ⇒ ⊗₁ λ⇒
+                      ≈ λ⇒
+                      ⟩
+
+    swapInner-unitʳ : ∀ {X Y} →
+                      [ (X ⊗₀ Y) ⊗₀ unit ⇒ (X ⊗₀ Y) ]⟨
+                        id ⊗₁ λ⇐   ⇒⟨ (X ⊗₀ Y) ⊗₀ (unit ⊗₀ unit) ⟩
+                        i⇒         ⇒⟨ (X ⊗₀ unit) ⊗₀ (Y ⊗₀ unit) ⟩
+                        ρ⇒ ⊗₁ ρ⇒
+                      ≈ ρ⇒
+                      ⟩
+
+  -- The interchange is a natural isomorphism.
+  
+  swapInner-naturalIso : ⊗ ∘F (⊗ ⁂ ⊗) ≃ ⊗ ∘F MonoidalProduct.⊗ M M
+  swapInner-naturalIso = niHelper (record
+    { η       = λ _ → i⇒
+    ; η⁻¹     = λ _ → i⇐
+    ; commute = λ _ → swapInner-natural
+    ; iso     = λ _ → swapInner.iso
+    })
+
+-- Shorthands for composing and inverting isomorphisms.
+
+open ≅ using () renaming (refl to idᵢ; sym to _⁻¹)
+infixr 9 _∘ᵢ_
+_∘ᵢ_ = λ {A B C} f g → ≅.trans {A} {B} {C} g f
+
+-- Braided monoidal categories have an interchange map.
+
+module BraidedInterchange {M : Monoidal C} (B : Braided M) where
+  open MonoidalReasoning M
+  open MonoidalUtilities M
+  open Braided B renaming (associator to α)
+  open Shorthands
+
+  -- Shorthands for braiding.
+
+  σ : ∀ {X Y} → X ⊗₀ Y ≅ Y ⊗₀ X
+  σ = braiding.FX≅GX 
+  module σ  {X Y} = _≅_ (σ {X} {Y})
```

Comment:
Oops, these were not supposed to be externally visible. Will fix. Maybe there's a point to making a `Shorthands` module for braided monoidal categories at some point, but this will be for another PR...

Reply:
Fixed.